iterator/cdc.go

```diff
@@ -29,224 +31,254 @@ import (
 	"gopkg.in/tomb.v2"
 )
 
-// CDCIterator iterates through the table's stream.
 type CDCIterator struct {
 	tableName     string
 	partitionKey  string
 	sortKey       string
 	streamsClient *dynamodbstreams.Client
-	cache         chan stypes.Record
+	cache         []recordWithTimestamp
+	cacheLock     sync.Mutex
 	streamArn     string
+	tomb          *tomb.Tomb
+	shards        map[string]*shardProcessor
+	lastPosition  position.Position
+	pollingPeriod time.Duration
+}
+
+type recordWithTimestamp struct {
```

Comment:
nitpick: it might be good to leave a todo for finding a prettier name and/or renaming the `Time` field (to show what's it about). We can do it in a separate PR. :+1: